pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -165,3 +176,49 @@ func (a Adapter) generateAndGetContainerConfig(componentName string, comp versio
 	containerConfig := a.Client.GenerateContainerConfig(*comp.Image, comp.Command, comp.Args, envVars, containerLabels)
 	return containerConfig
 }
+
+func (a Adapter) generateAndGetHostConfig() (container.HostConfig, error) {
+	portMap, err := getPortMap()
+	if err != nil {
+		return container.HostConfig{}, err
+	}
+	hostConfig := container.HostConfig{}
+	if len(portMap) > 0 {
+		hostConfig = a.Client.GenerateHostConfig(false, false, portMap)
+	}
+	return hostConfig, nil
+}
+
+func getPortMap() (nat.PortMap, error) {
```

Comment:
Need comments. Entire function has no comments on what's happening.

Reply:
added comment for the function